### PR TITLE
cc110x: header cleanups

### PR DESCRIPTION
**Pull request by gebart**
I stumbled across some empty `extern "C"` and C++ style comments in the cc110x headers. This PR removes the empty `extern "C"` and replaces the comments with C style comments (`/*  xxxx */`).

### CODE REVIEW

**Comment by BytesGalore** [comment]
Hi, removing the empty `extern "C"`s results in `file does not have a C++ compatible header: ` 
We decided to provide `extern "C"` in all RIOT headers, even empty ones, to prevent linking problems when compiling them with c++. 
This way we can be sure that if the linker fails on name mangling at c<->c++  its not "our fault"

No objections on the other changes :)

**Comment by BytesGalore** [comment]
sorry I forgot to say this check is invoked by travis using [this](https://github.com/RIOT-OS/RIOT/blob/master/dist/tools/externc/check.sh) script.

**Comment by gebart** [comment]
Deleted the extern "C" changes.

File: drivers/include/cc110x_legacy/cc110x-config.h [hunk 1]
```diff
@@ -71,30 +71,30 @@ typedef struct {
...
-    unsigned LL_ACK     : 1;    ///< Is set if Link-Level ACK is received, otherwise 0 (reset on new burst)
-    unsigned CAA        : 1;    ///< The status of the air (1 = air free, 0 = air not free)
-    unsigned CRC_STATE  : 1;    ///< The CRC status of last received packet (1 = OK, 0 = not OK)
-    unsigned SEQ        : 1;    ///< Sequence number (toggles between 0 and 1)
-    unsigned MAN_WOR    : 1;    ///< Manual WOR set (for randomized WOR times => no synch)
-    unsigned KT_RES_ERR : 1;    ///< A hwtimer resource error has occurred (no free timers available)
-    unsigned TX         : 1;    ///< State machine TX lock, only ACKs will be received
-    unsigned WOR_RST    : 1;    ///< Reset CC1100 real time clock (WOR) on next WOR strobe
+    uint32_t TOF;               /**< Time of flight of the last packet and last ACK */
+    timex_t  TOA;               /**< Time of packet arriveal */
```
Comment (fnack) [inline drivers/include/cc110x_legacy/cc110x-config.h]: Fix spelling

File: drivers/include/cc110x_legacy/cc110x-config.h [hunk 2]
```diff
@@ -71,30 +71,30 @@ typedef struct {
...
+    uint32_t TCP;               /**< Time to compute packet */
+    unsigned RPS        : 16;   /**< Raw packets sent to transmit last packet */
+    unsigned RETC       : 8;    /**< Retransmission count of last send packet */
```
Comment (fnack) [inline drivers/include/cc110x_legacy/cc110x-config.h]: send => sent

File: drivers/include/cc110x_legacy/cc110x-config.h [hunk 3]
```diff
@@ -71,30 +71,30 @@ typedef struct {
...
+    unsigned _RSSI      : 8;    /**< The RSSI value of last received packet */
+    unsigned RSSI_SEND  : 8;    /**< The RSSI value of the last send unicast packet of this node */
```
Comment (fnack) [inline drivers/include/cc110x_legacy/cc110x-config.h]: send => sent

**Comment by fnack** [comment]
Found just some spelling that can quickly be fixed with this PR as well. What I don't really like is that
```drivers/include/cc110x_legacy/cc110x-config.h
drivers/include/cc110x_legacy/cc110x-interface.h```
are still producing doxygen warnings. Would you like to invest a little more effort and make doxygen happy?

Side note: The legacy driver is supposed to be removed when all the older boards are ported to the new periph interfaces (in this case periph/gpio and periph/spi).

**Comment by gebart** [comment]
Corrected the spelling according to comments. I do not feel like it is worthwhile to spend more time on a deprecated driver, since it will be deleted soon-ish anyway.

**Comment by fnack** [comment]
Fine with me. ACK after you squashed!

**Pull request merged**